### PR TITLE
AGL Recipe Mailer

### PR DESCRIPTION
**Pull request by alex-latham**
Big notes:
* DOES NOT WORK ON PRODUCTION -- WE HATE SENDGRID (but Dione says that's okay)
* Background workers implemented for emailing a recipe to all of a users' friends
* Includes complete testing for mailer and job
* To use the background workers locally, run `redis-server` and `bundle exec sidekiq -C config/sidekiq.yml` in their own terminal windows in your knead_recipes repo

Rubocop: zero citations, 30 files
Coverage: 99.48% -- not sure how to test the User#share_with_friends method, and missing coverage in recipe_mailer_job.rb appears to be a result of how background workers get called (by using #perform_later rather than #perform).

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,6 +38,9 @@ gem 'faraday'
 gem 'bootstrap-sass'
 gem 'jquery-rails'
 gem 'coffee-script'
+gem 'sidekiq'
+gem 'sinatra'
+gem 'sendgrid-ruby'
```
Comment (mikez321) [inline Gemfile:43]: Yay sendgrid!

File: config/application.rb [hunk 2]
```diff
@@ -27,5 +27,6 @@ class Application < Rails::Application
 
     # Don't generate system test files.
     config.generators.system_tests = nil
+    config.active_job.queue_adapter = :sidekiq
```
Comment (mikez321) [inline config/application.rb:30]: Getting the BG workers in action is pretty sweet.  Nice work!

**Review by mikez321 — APPROVED** [review]
Aside from the commented lines and things we talked about everything else is looking great!

File: config/environments/production.rb [hunk 3]
```diff
@@ -88,4 +88,21 @@
...
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.smtp_settings = {
+    address: 'smtp.sendgrid.net',
+    user_name: 'apikey',
+    password: ENV['SENDGRID_API_KEY'],
+    domain: 'knead-recipes.herokuapp.com',
+    port: 587,
+    authentication: :plain,
+    enable_starttls_auto: true
+  }
```
Comment (DavidTTran) [inline config/environments/production.rb:107]: This is beyond finicky..

File: Gemfile.lock [hunk 4]
```diff
@@ -327,8 +347,11 @@ DEPENDENCIES
   rspec-rails
   rubocop-rails
   sass-rails (~> 5.0)
+  sendgrid-ruby
```
Comment (DavidTTran) [inline Gemfile.lock:350]: This gem is a lie

Comment (mikez321) [reply Gemfile.lock:350]: A real gem that one....

**Pull request merged**